core/01_core.js

```diff
@@ -127,6 +127,15 @@
     errorMap[className] = errorBuilder;
   }
 
+  function buildCustomError(className, message) {
```

Comment:
Hmm, actually. I would consider adding the OS code as a third parameter and doing the assignment here. On Rust side you could conditionally call the function with either two or three parameters to avoid creation of even the `v8:: Undefined` value.